recipes/gdal/all/conanfile.py

```diff
@@ -308,25 +321,31 @@ def requirements(self):
         if self.options.with_cryptopp:
             self.requires("cryptopp/8.2.0")
         if self.options.with_crypto:
-            self.requires("openssl/1.1.1h")
+            self.requires("openssl/1.1.1i")
         # if not self.options.without_lerc:
         #     self.requires("lerc/2.1") # TODO: use conan recipe (not possible yet because lerc API is broken for GDAL)
         if self.options.get_safe("with_exr"):
-            self.requires("openexr/2.5.2")
-        # if self.options.get_safe("with_heif"):
-        #     self.requires("libheif/1.9.1")
+            self.requires("openexr/2.5.3")
+        if self.options.get_safe("with_heif"):
+            self.requires("libheif/1.9.1")
 
     def _validate_dependency_graph(self):
         if self.options.with_qhull and self.options["qhull"].reentrant:
             raise ConanInvalidConfiguration("gdal depends on non-reentrant qhull.")
+        if self.options.with_mongocxx:
+            mongocxx_version = tools.Version(self.deps_cpp_info["mongo-cxx-driver"].version)
+            if mongocxx_version < "3.0.0":
+                # TODO: handle mongo-cxx-driver v2
+                raise ConanInvalidConfiguration("gdal with mongo-cxx-driver < 3.0.0 not yet supported in this recipe.")
+            elif mongocxx_version < "3.4.0":
+                raise ConanInvalidConfiguration("gdal with mongo-cxx-driver v3 requires 3.4.0 at least.")
 
     def build_requirements(self):
         if self.settings.compiler != "Visual Studio":
+            self.build_requires("libtool/2.4.6")
+            self.build_requires("pkgconf/1.7.3")
             if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
                 self.build_requires("msys2/20200517")
```

Comment:
@madebr with this modification (and `self.run("{} -fiv".format(tools.get_env("AUTORECONF")), win_bash=tools.os_info.is_windows)`), autoreconf now fails on my mingw@windows:

```
gdal/3.2.1: Calling build()
gdal/3.2.1: run_in_windows_bash: C:\Users\spaceim\.conan_short\da178f\1\bin\usr\bin\bash.exe --login -c ^"cd \^"/c/users/spaceim/.conan/data/gdal/3.2.1/_/_/build/22bb066d4c0cb5a4242d454a87aa31899b1d43c2/source_subfolder\^" ^&^& PATH=\^"/c/users/spaceim/.conan/data/libtool/2.4.6/_/_/package/03bdfb27ed0d887115be62c7739000f272a68bd9/bin:/c/users/spaceim/.conan/data/pkgconf/1.7.3/_/_/package/c6817f477abe7e9a917b102f37dc1fd0c2d95f50/bin:/c/users/spaceim/.conan_short/da178f/1/bin/usr/bin:/c/users/spaceim/.conan/data/libiconv/1.16/_/_/package/c6817f477abe7e9a917b102f37dc1fd0c2d95f50/bin:/c/users/spaceim/.conan/data/qhull/8.0.1/_/_/package/c6817f477abe7e9a917b102f37dc1fd0c2d95f50/bin:/c/users/spaceim/.conan/data/automake/1.16.3/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin:/c/users/spaceim/.conan/data/proj/7.2.1/_/_/package/f40fac8c96b53927f2e6eec85220355162fe1b1b/bin:/c/users/spaceim/.conan/data/autoconf/2.69/_/_/package/3475bd55b91ae904ac96fde0f106a136ab951a5e/bin:/c/users/spaceim/.conan/data/sqlite3/3.34.0/_/_/package/ea0d7183af566ea84394b5e26b1e5e6eec97d6d7/bin:/c/users/spaceim/.conan/data/m4/1.4.18/_/_/package/03907e48899e7b467b3650dc72869dac61405c35/bin:/c/users/spaceim/.conan/data/jbig/20160605/_/_/package/0ad6bb798f5b8e965eeba2944aa0dd033a6c1dc0/bin:$PATH\^" ^&^& /c/users/spaceim/.conan/data/autoconf/2.69/_/_/package/3475bd55b91ae904ac96fde0f106a136ab951a5e/bin/autoreconf.exe -fiv ^"
autoreconf.exe: Entering directory `.'
autoreconf.exe: configure.ac: not using Gettext
autoreconf.exe: running: /c/users/spaceim/.conan/data/automake/1.16.3/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin/aclocal.exe --force
aclocal.exe: error: couldn't open directory '/c/users/spaceim/.conan/data/libtool/2.4.6/_/_/package/03bdfb27ed0d887115be62c7739000f272a68bd9/bin/share/aclocal;/c/users/spaceim/.conan/data/pkgconf/1.7.3/_/_/package/c6817f477abe7e9a917b102f37dc1fd0c2d95f50/bin/aclocal': No such file or directory
autoreconf.exe: /c/users/spaceim/.conan/data/automake/1.16.3/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin/aclocal.exe failed with exit status: 1
gdal/3.2.1:
gdal/3.2.1: ERROR: Package '22bb066d4c0cb5a4242d454a87aa31899b1d43c2' build failed
gdal/3.2.1: WARN: Build folder C:\Users\spaceim\.conan\data\gdal\3.2.1\_\_\build\22bb066d4c0cb5a4242d454a87aa31899b1d43c2
ERROR: gdal/3.2.1: Error in build() method, line 712
        self.run("{} -fiv".format(tools.get_env("AUTORECONF")), win_bash=tools.os_info.is_windows)
        ConanException: Error 1 while executing /c/users/spaceim/.conan/data/autoconf/2.69/_/_/package/3475bd55b91ae904ac96fde0f106a136ab951a5e/bin/autoreconf.exe -fiv
```

Any idea?

Reply:
It looks like the `configure.ac` includes gettext m4 macros (supplied by libgettext)
I don't know whether our gettext cci recipe supplies these macros.
The file should be gettext.m4 or similar.

It can be fixed by adding `gettext.m4` to the libgettext package and adding the location of `gettext.m4` to libgettext's [`ACLOCAL_PATH` environment variable](https://www.gnu.org/software/automake/manual/html_node/Macro-Search-Path.html).
As last, `gettext` must be added to the build_requirements of gdal.

Reply:
`gettext.m4` and `iconv.m4` are installed in `gettext` recipe, but removed in `package()`

Reply:
I tried to add `gettext/0.20.1` (modified with https://github.com/conan-io/conan-center-index/pull/4462) to build_requirements of gdal recipe.

On Macos autoreconf says:

```
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: /Users/spaceim/.conan/data/automake/1.16.3/_/_/package/3c28ba69276b46fd46e9c2060ba991d725ccab43/bin/aclocal --force
autoreconf: configure.ac: tracing
autoreconf: running: /Users/spaceim/.conan/data/libtool/2.4.6/_/_/package/f82f59608a22017ca5ceea2114611456d4373a12/bin/libtoolize --copy --force
libtoolize: putting auxiliary files in '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
libtoolize: Consider adding '-I m4' to ACLOCAL_AMFLAGS in Makefile.am.
libtoolize: 'AC_PROG_RANLIB' is rendered obsolete by 'LT_INIT'
autoreconf: running: /Users/spaceim/.conan/data/autoconf/2.69/_/_/package/cc471ed355602ffa7007968c83aa4e7e648a38b5/bin/autoconf --force
autoreconf: running: /Users/spaceim/.conan/data/autoconf/2.69/_/_/package/cc471ed355602ffa7007968c83aa4e7e648a38b5/bin/autoheader --force
autoreconf: configure.ac: not using Automake
autoreconf: Leaving directory `.'
```